lumeer-storage/lumeer-storage-mongodb/src/main/java/io/lumeer/storage/mongodb/dao/system/MongoUserNotificationDao.java

```diff
@@ -0,0 +1,107 @@
+ */
+package io.lumeer.storage.mongodb.dao.system;
+
+import static io.lumeer.storage.mongodb.util.MongoFilters.idFilter;
+
+import io.lumeer.api.model.UserNotification;
+import io.lumeer.storage.api.dao.UserNotificationDao;
+import io.lumeer.storage.api.exception.StorageException;
+import io.lumeer.storage.mongodb.codecs.UserNotificationCodec;
+
+import com.mongodb.MongoException;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.model.Filters;
+import com.mongodb.client.model.FindOneAndReplaceOptions;
+import com.mongodb.client.model.IndexOptions;
+import com.mongodb.client.model.Indexes;
+import com.mongodb.client.model.ReturnDocument;
+import com.mongodb.client.model.Sorts;
+import org.bson.Document;
+import org.bson.conversions.Bson;
+
+import java.util.ArrayList;
+import java.util.List;
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+
+/**
+ * @author <a href="mailto:marvenec@gmail.com">Martin Večeřa</a>
+ */
+@ApplicationScoped
+public class MongoUserNotificationDao extends SystemScopedDao implements UserNotificationDao {
+
+   public static final String COLLECTION_NAME = "userNotifications";
+
+   @PostConstruct
+   public void initDb() {
+      createUserNotificationsRepository();
+   }
+
+   @Override
+   public List<UserNotification> getRecentNotifications(final String userId) {
+      Bson idFilter = Filters.eq(UserNotificationCodec.USER_ID, userId);
+
+      return databaseCollection()
+            .find(idFilter)
+            .limit(500)
+            .sort(Sorts.descending(UserNotificationCodec.CREATED_AT))
+            .into(new ArrayList<>());
+   }
+
+   @Override
+   public UserNotification getNotificationById(final String notificationId) {
+      return databaseCollection().find(idFilter(notificationId)).first();
+   }
+
+   @Override
+   public UserNotification updateNotification(final UserNotification notification) {
+      FindOneAndReplaceOptions options = new FindOneAndReplaceOptions().returnDocument(ReturnDocument.AFTER).upsert(true);
+      try {
+         UserNotification returnedNotification = databaseCollection().findOneAndReplace(idFilter(notification.getId()), notification, options);
+         if (returnedNotification == null) {
+            throw new StorageException("Notification '" + notification.getId() + "' has not been updated.");
+         }
+         // TODO send push notification
+         return returnedNotification;
+      } catch (MongoException ex) {
+         throw new StorageException("Cannot update notification " + notification, ex);
+      }
+   }
+
+   @Override
+   public List<UserNotification> createNotificationsBatch(final List<UserNotification> notifications) {
+      databaseCollection().insertMany(notifications);
+      return notifications;
+   }
+
+   @Override
+   public void createUserNotificationsRepository() {
+      if (database.getCollection(COLLECTION_NAME) == null) {
```

Comment:
This is not correct. You see there warning `Condition .... is always false`.
Why you do not look at existing solution in other SystemScopedDao? For example: `MongoOrganizationDao`

Reply:
Why I do not look? Well, the answer is simple, I did not know what I do not know. Then it is hard to look anywhere...

Reply:
But! I copied this from some other Dao! I need to find it!

Reply:
It was on 3 other places. I fixed it! Thanks for spotting this!